### PR TITLE
Reorder the user imports fields

### LINKED ISSUE #3719: Import pages: adjustments needed because of membership number
**Description:** 
The new member number attribute has changed the referencing priority in the import. This should be reflected on the import pages and in the csv files.

**What should happen:**

1. Account import page
Changed column sequence:
title; first_name; last_name; email; member_number; default_vote_weight; gender; pronoun; username; default_password; is_active; is_physical_person; saml_id
This order should be integrated in the description on the page and in the 'Download CSV example File'. 
Addtionally the export of the Account csv file should have this order.

2. Participant import:
Changed column sequence in csv file: 
title; first_name; last_name; email; member_number; structure_level; groups; number; vote_weight; gender; pronoun; username; default_password; is_active; is_physical_person; is_present; saml_id; comment
This order should be integrated in the description on the page and in the 'Download CSV example File'. 
Addtionally the export of the participant csv file should have this order.

### PR DESCRIPTION
**Pull request by reiterl**
Resolve #3719

### CODE REVIEW

**Review by Elblinator — CHANGES_REQUESTED** [review]
Add-on: 
Please also change the order in the previews 

In the account-import:
- swap pronoun and gender

**Comment by reiterl** [comment]
Added the swap. I think the order of the fields in the preview is delivered by the backend.

**Review by Elblinator — APPROVED** [review]

**Review by emanuelschuetze — APPROVED** [review]

**Review by Elblinator — CHANGES_REQUESTED** [review]
As far as I can see the file src/app/domain/models/users/user.constants now is not used anywhere and should be deleted

**Review by luisa-beerboom — CHANGES_REQUESTED** [review]
I am against separating these two header definitions, it's unnecessarily shaky, it would be better to implement a way to assign these properties a client-side weight separately and then do some column sorting in the import-list components. I'd suggest do it via the defaultColumns input of the import-list.component.ts

**Comment by reiterl** [comment]
Adding weight to the datastructure would add complexity to the backend import component interface. And the interface would be uglier. I don't think @luisa-beerboom  's idea is a good idea. Having two definitions for two different imports should be no problem in my opinion. I think my solution is fine. @rrenkert  what do you think?

**Comment by rrenkert** [comment]
The easiest way would be to define two const lists of `userHeadersAndVerboseNames`.
Currently `userHeadersAndVerboseNames` is only used for the mentioned imports and defining two lists will allow separate sorting of the columns for each of the imports without the overhead of additional attributes.

**Comment by reiterl** [comment]
Okay, I did that in my PR. I will remove the surplus `userHeadersAndVerboseNames`, check it and then I think my PR is okay so far.

**Review by luisa-beerboom — CHANGES_REQUESTED** [review]
Like I said: Don't transform the `HeadersAndVerboseNames` objects. They have the expressed purpose of managing the header names and should be as unified as possible for the sake of consistency.

There are already client-side column configurations (see f.E. the `AccountImportListComponent`:  It's an attribute called `columns`) add a new optional field `weight` to those and make the `BaseViaBackendImportListComponent` sort the columns by that if it is there.

File: client/src/app/site/pages/meetings/pages/participants/export/participant-csv-export.service/participant-csv-export.service.ts [hunk 1]
```diff
@@ -40,12 +43,16 @@ export class ParticipantCsvExportService {
     //     ]
     // ]);
 
+    public columns = Object.keys(participantHeadersAndVerboseNames).sort(
+        (a, b) => participantColumnsWeight[a] - participantColumnsWeight[b]
+    );
```
Comment (luisa-beerboom) [inline client/src/app/site/pages/meetings/pages/participants/export/participant-csv-export.service/participant-csv-export.service.ts:48]: If `participantColumnsWeight` already contains all relevant fields, why even import `participantHeadersAndVerboseNames` at all

File: client/src/app/site/pages/meetings/pages/participants/pages/participant-import/definitions/index.ts [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
+    gender: 10,
+    pronoun: 11,
+    username: 12,
+    default_password: 13,
+    is_active: 14,
+    is_physical_person: 15,
+    is_present: 16,
+    saml_id: 17,
+    comment: 18
+};
```
Comment (luisa-beerboom) [inline client/src/app/site/pages/meetings/pages/participants/pages/participant-import/definitions/index.ts:33]: Using an array with the index being the weight may also work, would spare you the work of having to re-sort the keys every time you need the keys in-order

File: client/src/app/site/pages/meetings/pages/participants/pages/participant-import/components/participant-import-list/participant-import-list.component.ts [hunk 3]
```diff
@@ -12,9 +12,11 @@ import { ParticipantImportService } from '../../services';
     styleUrls: [`./participant-import-list.component.scss`]
 })
 export class ParticipantImportListComponent extends BaseViaBackendImportListMeetingComponent {
-    public possibleFields = Object.keys(participantHeadersAndVerboseNames);
+    public possibleFields = Object.keys(participantHeadersAndVerboseNames).sort(
+        (a, b) => participantColumnsWeight[a] - participantColumnsWeight[b]
+    );
```
Comment (luisa-beerboom) [inline client/src/app/site/pages/meetings/pages/participants/pages/participant-import/components/participant-import-list/participant-import-list.component.ts:17]: see comments in `participant-csv-export.service.ts` and `participant-import/definitions/index.ts`

File: client/src/app/gateways/export/csv-export.service/csv-export-for-backend.service.spec.ts [hunk 4]
```diff
@@ -114,7 +114,7 @@ describe(`CsvExportForBackendService`, () => {
 
     it(`test dummy export method with default settings`, () => {
         service.dummyCSVExport(
-            { proper: `Fancy`, tea: `Tea`, strength: `Muscle` },
+            Object.keys({ proper: `Fancy`, tea: `Tea`, strength: `Muscle` }),
```
Comment (luisa-beerboom) [inline client/src/app/gateways/export/csv-export.service/csv-export-for-backend.service.spec.ts:117]: Just use the list of keys at this point, no use keeping this as an object

**Review by luisa-beerboom — CHANGES_REQUESTED** [review]
This is much better, some minor improvements though

File: client/src/app/site/pages/meetings/pages/participants/export/participant-csv-export.service/participant-csv-export.service.ts (hunk 1 above)
Comment (reiterl) [reply client/src/app/site/pages/meetings/pages/participants/export/participant-csv-export.service/participant-csv-export.service.ts:48]: yes, changed it.

File: client/src/app/site/pages/meetings/pages/participants/pages/participant-import/definitions/index.ts (hunk 2 above)
Comment (reiterl) [reply client/src/app/site/pages/meetings/pages/participants/pages/participant-import/definitions/index.ts:33]: Now use an array.

File: client/src/app/site/pages/meetings/pages/participants/pages/participant-import/components/participant-import-list/participant-import-list.component.ts (hunk 3 above)
Comment (reiterl) [reply client/src/app/site/pages/meetings/pages/participants/pages/participant-import/components/participant-import-list/participant-import-list.component.ts:17]: Done.

File: client/src/app/gateways/export/csv-export.service/csv-export-for-backend.service.spec.ts (hunk 4 above)
Comment (reiterl) [reply client/src/app/gateways/export/csv-export.service/csv-export-for-backend.service.spec.ts:117]: done.

**Comment by reiterl** [comment]
Updated: Use lists for the order config. Remove surplus import, no longer sort and simplify one test.

**Review by luisa-beerboom — APPROVED** [review]

**Review by Elblinator — APPROVED** [review]

**Comment by Elblinator** [comment]
please solve merge conflicts

**Pull request merged**